Source/monster.cpp

```diff
@@ -6080,46 +6080,45 @@ BOOL PosOkMonst2(int i, int x, int y)
 
 BOOL PosOkMonst3(int i, int x, int y)
 {
-	int oi, mi, j, objtype;
+	int j, oi, objtype, mi;
 	BOOL ret, fire, isdoor;
 
 	fire = FALSE;
+	ret = TRUE;
 	isdoor = FALSE;
-	if (dObject[x][y]) {
+
+	if(ret && dObject[x][y] != 0) {

```

Comment:
I guess there is a debug or early part of the code missing here that could set ret?

Reply:
`ret` is set to `TRUE` in the beginning so it's always valid, but gets optimized away. You can see the check is there in the beta.

Reply:
Guess it's just bad code, was hoping there was some forgotten secret :)